pwiz_tools/Shared/Common/Chemistry/MoleculeMassOffset.cs

```diff
@@ -18,106 +18,489 @@
  */
 using System;
 using System.Collections.Generic;
+using System.Collections.ObjectModel;
 using System.Globalization;
 using System.Linq;
 using System.Text;
-using pwiz.Common.SystemUtil;
+using pwiz.Common.Collections;
 
 namespace pwiz.Common.Chemistry
 {
     /// <summary>
     /// Holds a chemical formula as well as an extra offset for the monoisotopic mass and the average mass.
     /// This is used when a molecule is made up of some things that we know the chemical formula for,
     /// and other things where the user has only told us the mono and average masses.
+    /// Examples of these formulas with mass offsets include crosslinking and Hardklor output.
     /// </summary>
-    public class MoleculeMassOffset : Immutable, IFormattable
+    public class MoleculeMassOffset : Molecule, IEquatable<MoleculeMassOffset>, IComparable<MoleculeMassOffset>
```

Comment:
I don't think it's a good idea to have "MoleculeMassOffset" be a subclass of "Molecule".
This seems to be a case where the relationship really is "has a" not "is a".
One of the downsides to having "MoleculeMassOffset" being a subclass of "Molecule" is that it becomes impossible to declare in the code that something is supposed to be a pure molecule with no offset.

The other thing is that with the inheritance this way it's really difficult to implement the method "Molecule.Plus" correctly.

"MoleculeMassOffset" overrides "Molecule.Plus", but that then means that if you have a Molecule "m" and a MoleculeMassOffset "mmo", you end up with the situation where "m.Plus(mmo)" is very different than "mmo.Plus(m)".